### PR TITLE
Paddle OCR with ERNIEKit

### PR DESCRIPTION
**Pull request by forBlank**
update gradient_accumulation_steps*8 in paddleocr_vl_sft config for single GPU SFT

### CODE REVIEW

File: examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml [hunk 1]
```diff
@@ -44,9 +44,9 @@ save_total_limit: 5
 save_strategy: steps
 logging_steps: 1
 release_grads: True
-gradient_accumulation_steps: 1
-logging_dir: ./log
-output_dir: ./output
+gradient_accumulation_steps: 8
+logging_dir: ./Paddle-OCR-VL-SFT-Bengali-log
```
Comment (nepeplwu) [inline examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml]:
```suggestion
logging_dir: ./PaddleOCR-VL-SFT-Bengali/tensorboard_logs/
```

File: examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml [hunk 2]
```diff
@@ -44,9 +44,9 @@ save_total_limit: 5
...
+output_dir: ./Paddle-OCR-VL-SFT-Bengali
```
Comment (nepeplwu) [inline examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml]:
```suggestion
output_dir: ./PaddleOCR-VL-SFT-Bengali
```

File: docs/paddleocr_vl_sft_zh.md [hunk 3]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
 ```
-- 我们提供了[孟加拉语测试数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-test_Bengali.jsonl)，可用于推理来观察微调效果。
 
-### 超参数说明
+## 训练配置
 
-- `max_steps`：训练总步数, 约等于 `(D × E) / (G × B × A)`。
+我们针对孟加拉语示例数据集提供了[配置文件](../examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml)，其中的关键训练超参数如下：
+
+- `max_steps=926`：训练总步数, 约等于 `(D × E) / (G × B × A)`。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md:55]: 可以结合我们的示例说明下情况，例如孟加拉语的数据集量级多大，我们准备用单卡训练，训练n个epoch（为什么是n个，数据集如果比较困难，需要多学几个epoch吗）

关于G的解释，就写 `用于训练的GPU数量` 吧
关于B的解释，不要引入打包大小的概念，直接说 `单卡的训练batchsize`
关于A的解释，说明下为什么要做梯度累计，一般怎么调整

File: docs/paddleocr_vl_sft_zh.md [hunk 4]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
     - `G`：数据并行的 GPU 数目。
     - `B`：每 GPU 每步的批大小 (打包大小)。
     - `A`：梯度累积的步数。
-- `warmup_steps`：线性预热步数, 建议设置成最大步数的 1% `0.01 × max_steps`。
-- `packing_size`：单个序列中打包的样本数目。
-- `padding`：设置为 `False` 以避免将序列填充到最大序列长度。
-- `max_seq_len`：确保该值大于输入样本的最大序列长度。
\ No newline at end of file
+- `warmup_steps=10`：线性预热步数, 建议设置成最大步数的 1% `0.01 × max_steps`。
+- `packing=True`：设置为 `True` 以将批内的样本打包成序列。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]: 是否需要暴露让用户感知？如果默认我们就建议设置为packing，这里就不要暴露了

File: docs/paddleocr_vl_sft_zh.md [hunk 5]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+- `packing_size=8`：单个序列中打包的样本数目。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]: 同上

File: docs/paddleocr_vl_sft_zh.md [hunk 6]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+- `gradient_accumulation_steps=8`：梯度累积步数。
+- `padding=False`：设置为 `False` 以避免将序列填充到最大序列长度。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]: 同上

File: docs/paddleocr_vl_sft_zh.md [hunk 7]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+- `max_seq_len=16384`：确保该值大于输入样本的最大序列长度。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]: 需要一个简单的说明

File: docs/paddleocr_vl_sft_zh.md [hunk 8]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+python -m pip install paddlex
+python -m pip install https://paddle-whl.bj.bcebos.com/nightly/cu126/safetensors/safetensors-0.6.2.dev0-cp38-abi3-linux_x86_64.whl
+```
+
+### 推理模型准备
+
+从 PaddleOCR-VL 中拷贝必要的推理配置文件到 SFT 训练完成后保存的模型目录中
+
+```
+cp PaddlePaddle/PaddleOCR-VL/chat-template.jinja Paddle-OCR-VL-SFT-Bengali-log
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
cp PaddlePaddle/PaddleOCR-VL/chat_template.jinja PaddleOCR-VL-SFT-Bengali
```

File: docs/paddleocr_vl_sft_zh.md [hunk 9]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+cp PaddlePaddle/PaddleOCR-VL/inference.yaml Paddle-OCR-VL-SFT-Bengali-log
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
cp PaddlePaddle/PaddleOCR-VL/inference.yml PaddleOCR-VL-SFT-Bengali
```

File: docs/paddleocr_vl_sft_zh.md [hunk 10]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
 
 ```
 huggingface-cli download PaddlePaddle/PaddleOCR-VL --local-dir PaddlePaddle/PaddleOCR-VL
 ```
 
-- 我们提供了[孟加拉语训练数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-train_Bengali.jsonl)可用于微调 PaddleOCR-VL-0.9B 对孟加拉语进行识别，或者你可以按照 [SFT VL 数据集格式](./datasets.md#sft-vl-dataset) 来构建自己的微调数据集。
-- 使用命令行和 YAML 配置文件来启动训练：
+### 数据集准备
+
+可以按照 [SFT VL 数据集格式](./datasets.md#sft-vl-dataset) 来构建微调数据集，同时我们也提供了一个快速上手的[孟加拉语训练数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-train_Bengali.jsonl)，可用于微调 PaddleOCR-VL-0.9B 对孟加拉语进行识别，使用以下命令下载：
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
请参考 [SFT VL 数据集格式](./datasets.md#sft-vl-dataset) 来构建微调数据集。为了方便起见，我们也提供了一个快速上手的[孟加拉语训练数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-train_Bengali.jsonl)，可用于微调 PaddleOCR-VL-0.9B 对孟加拉语进行识别，使用以下命令下载：
```

File: docs/paddleocr_vl_sft_zh.md [hunk 11]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
-- `padding`：设置为 `False` 以避免将序列填充到最大序列长度。
-- `max_seq_len`：确保该值大于输入样本的最大序列长度。
\ No newline at end of file
+- `warmup_steps=10`：线性预热步数, 建议设置成最大步数的 1% `0.01 × max_steps`。
+- `packing=True`：设置为 `True` 以将批内的样本打包成序列。
+- `packing_size=8`：单个序列中打包的样本数目。
+- `gradient_accumulation_steps=8`：梯度累积步数。
+- `padding=False`：设置为 `False` 以避免将序列填充到最大序列长度。
+- `max_seq_len=16384`：确保该值大于输入样本的最大序列长度。
+- `learning_rate=1e-5`：学习率，即每次参数更新的幅度。
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]: 学习率看起来和配置文件对不上？

File: docs/paddleocr_vl_sft_zh.md [hunk 12]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+ERNIEKit 默认使用机器上的全部 GPU，可以通过环境变量 `CUDA_VISIBLE_DEVICES` 设置 ERNIEKit 能够使用的 GPU。
+
+GPU 的数目 `GPU_num` 会影响训练超参数 `learning_rate & packing_size & gradient_accumulation_steps` 配置。理论上，每个更新步使用的样本数目 `sample_num = G*B*A`，近似与学习率 `learning_rate` 成正线形关系，因此，当 GPU 数目增加 `N` 倍变为 `N*GPU` 时，有两种调整方式：
+1. 保持 `sample_num` 不变
+    - 将 `packing_size` 减少 `x` 倍，变成 `packing_size/x`
+    - 将 `gradient_accumulation_steps` 减少 `y` 倍，变成 `gradient_accumulation_steps/y` 
+    - 满足 `x*y = N` 即可
+2. 将 `learning_rate` 增加 `N` 倍，变成 `N*learning_rate`
+
+可以通过 `tensorboard` 对训练过程可视化，使用以下命令行即可启动（默认端口 port 为 `8084`，需要根据实际情况设置可用端口）：
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
可以通过 `tensorboard` 对训练过程可视化，使用以下命令行即可启动（下方命令将端口 port 为 `8084`，需要根据实际情况设置可用端口）：
```

File: docs/paddleocr_vl_sft_zh.md [hunk 13]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+
+```
+pip install tensorboard
+tensorboard --logdir ./Paddle-OCR-VL-SFT-Bengali-log --port 8084
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
tensorboard --logdir ./PaddleOCR-VL-SFT-Bengali --port 8084 --host `hostname -i`
```

File: docs/paddleocr_vl_sft_zh.md [hunk 14]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+```
+
+成功启动后该服务后，在浏览器输入 `ip:port` ，则可以看到训练日志（通过 `hostname -i` 命令可以查看机器的 ip 地址）
+
+## 模型结构说明
+
+训练结束后，模型会保存在 `output_dir=./Paddle-OCR-VL-SFT-Bengali` 指定路径下，其中包含：
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
训练结束后，模型会保存在 `output_dir=./PaddleOCR-VL-SFT-Bengali` 指定路径下，其中包含：
```

File: docs/paddleocr_vl_sft_zh.md [hunk 15]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
 这时，就需要通过 SFT (Supervised Fine-Tuning) 来提升模型的准确性和鲁棒性。
 
 
-## SFT 训练
+## 环境配置
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
## 1. 环境配置
```

File: docs/paddleocr_vl_sft_zh.md [hunk 16]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+
+请参考 [ERNIEKit-安装文档](./erniekit.md#2-installation) 中的安装步骤对训练环境进行配置。
+
+## 模型和数据集准备
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
## 2. 模型和数据集准备
```

File: docs/paddleocr_vl_sft_zh.md [hunk 17]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
+
+### 模型准备
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
### 2.1 模型准备
```

File: docs/paddleocr_vl_sft_zh.md [hunk 18]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
-- 在 [huggingface](https://huggingface.co/PaddlePaddle/PaddleOCR-VL/tree/main/PaddleOCR-VL-0.9B) 或者 [modelscope](https://modelscope.cn/models/PaddlePaddle/PaddleOCR-VL/files) 可以下载 PaddleOCR-VL-0.9B 模型。
+在 [huggingface](https://huggingface.co/PaddlePaddle/PaddleOCR-VL/tree/main/PaddleOCR-VL-0.9B) 或者 [modelscope](https://modelscope.cn/models/PaddlePaddle/PaddleOCR-VL/files) 可以下载 PaddleOCR-VL-0.9B 模型。
 
 ```
 huggingface-cli download PaddlePaddle/PaddleOCR-VL --local-dir PaddlePaddle/PaddleOCR-VL
 ```
 
-- 我们提供了[孟加拉语训练数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-train_Bengali.jsonl)可用于微调 PaddleOCR-VL-0.9B 对孟加拉语进行识别，或者你可以按照 [SFT VL 数据集格式](./datasets.md#sft-vl-dataset) 来构建自己的微调数据集。
-- 使用命令行和 YAML 配置文件来启动训练：
+### 数据集准备
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
### 2.2 数据集准备
```

File: docs/paddleocr_vl_sft_zh.md [hunk 19]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
 ```
-erniekit train examples/configs/PaddleOCR-VL/sft/run_ocr_vl_sft_16k.yaml \
-        model_name_or_path=PaddlePaddle/PaddleOCR-VL \
-        train_dataset_path=/PAHT/TO/DATASET \
+wget https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-train_Bengali.jsonl
 ```
-- 我们提供了[孟加拉语测试数据集](https://paddleformers.bj.bcebos.com/datasets/ocr_vl_sft-test_Bengali.jsonl)，可用于推理来观察微调效果。
 
-### 超参数说明
+## 训练配置
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
## 3. 训练配置
```

File: docs/paddleocr_vl_sft_zh.md [hunk 20]
```diff
@@ -26,32 +26,158 @@ PaddleOCR-VL 是一款为文档解析任务量身打造的、性能顶尖 (SOTA)
...
\ No newline at end of file
+- `warmup_steps=10`：线性预热步数, 建议设置成最大步数的 1% `0.01 × max_steps`。
+- `packing=True`：设置为 `True` 以将批内的样本打包成序列。
+- `packing_size=8`：单个序列中打包的样本数目。
+- `gradient_accumulation_steps=8`：梯度累积步数。
+- `padding=False`：设置为 `False` 以避免将序列填充到最大序列长度。
+- `max_seq_len=16384`：确保该值大于输入样本的最大序列长度。
+- `learning_rate=1e-5`：学习率，即每次参数更新的幅度。
+
+## SFT 训练
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft_zh.md]:
```suggestion
## 4. SFT 训练
```
